src/csv_to_confluence.py

```diff
@@ -833,14 +862,49 @@ def upload_summary_for_confluence_pages(self):
         node_ids = self._mapper.get_all_keys()
         tbody = self.__summary.create_empty_summary_table()
         for node_id in node_ids:
-            commonname = self._mapper.get_node_value_from_mapping_dict(node_id, 'COMMON_NAME')
-            if self._confluence.does_page_exists(commonname):
-                page = self._confluence.get_page_content(commonname)
-                row = self.__summary.create_summary_table_row_from_confluence_page(commonname, page)
+            common_name = self._mapper.get_node_value_from_mapping_dict(node_id, 'COMMON_NAME')
+            if self._confluence.does_page_exists(common_name):
+                page = self._confluence.get_page_content(common_name)
+                row = self.__summary.create_summary_table_row_from_confluence_page(common_name, page)
                 tbody.find('tbody').append(row)
+
+        file_path = self.__create_error_rate_histogram_image()
+        row = self.__summary.create_histogram_table_row(row, self._confluence_parent_page, file_path)
+        tbody.find('tbody').append(row)
         table = self.__summary.create_summary_table_frame()
         table.append(tbody)
         self._confluence.update_confluence_page(self._confluence_parent_page, str(table))
+        self.__delete_char_file(file_path)
+
+    def __create_error_rate_histogram_image(self):
+        """
+        This method collects statistical data from each node and uses them to generate a histogram
+        """
+        node_ids = self._mapper.get_all_keys()
+        valid_paths = []  # List of paths leading to newest data file of each node
+        year = datetime.now().year
+        data_man = DataManager(self.__handler._confluence.get_confluence())
+        for node_id in node_ids:
+            common_name = self._mapper.get_node_value_from_mapping_dict(node_id, 'COMMON_NAME')
+            filename = f"{node_id}_stats_{year}.csv"
+            path = data_man.get_stat_file_from_page(
+                self.__handler._confluence.get_confluence().get_page_id(self.__space, common_name), filename)
```

Comment:
Do following instead:

```python
self.__csv_handler=InfoCSVHandler()
name_csv = self.__csv_handler.generate_node_csv_name(node_id)
path_csv = os.path.join(working_dir, name_csv)
```

Reply:
I changed 
```
path_csv = os.path.join(working_dir, name_csv)
```
to
```
path_csv = os.path.join(working_dir, node_id, name_csv)
```
because i assumed the structure of the project looks like this:
-broker-monitor
----src
----1
--------1_stats_2024.csv
--------1_errors_2014.csv
----2
--------2_stats_2024.csv
...

If that is not the case please correct.